app/routes/task_routes.py

```diff
@@ -0,0 +1,161 @@
+from flask import abort, Blueprint, jsonify, make_response, request
+from app import db
+from app.models.task import Task
+from datetime import datetime
+import requests
+import logging
+import os
+slack_token = os.environ.get("SLACK_BOT_TOKEN")
+logger = logging.getLogger(__name__)
+
+task_bp = Blueprint("task", __name__, url_prefix="/tasks")
+
+# Add a task
+@task_bp.route("", methods=["POST"])
+def add_task():
+    request_body = request.get_json()
+    verify_task_inputs(request_body)
+    new_task = Task.from_dict(request_body)
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    response_dict = message_for_only_one_task(new_task)
+
+    return make_response(response_dict, 201)
+
+# Get all tasks and the user can determine what the tasks
+# should be sorted by using the query params. Accepted sort
+# by functions is by ID and by title
+@task_bp.route("", methods=["GET"])
+def get_tasks():
+    sort_query = request.args.get("sort", default=None)
+    sort_by_query = request.args.get("sort_by", default="title")
+
+    all_tasks = Task.query.all()
+
+    response_list = [task.to_dict() for task in all_tasks]
+
+    if sort_by_query == "title":
+        sort_key = get_title 
+    elif sort_by_query == "id":
+        sort_key = get_id
+    elif sort_by_query:
+        abort(make_response({"details": "sort type is not accepted"}, 400))
+    if sort_query == "asc":
+        response_list.sort(key=sort_key)
+    elif sort_query == "desc":
+        response_list.sort(reverse=True, key=sort_key)
+    elif sort_query:
+        abort(make_response({"details": "sort must be ascending or descending"}, 400))
+
+    return jsonify(response_list), 200
+
+# Get the information associated with one task. The User
+# identifies the specific task with the id in the URL.
+@task_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    task = verify_item(Task, task_id)
+
+    response_dict = message_for_only_one_task(task)
+
+    return make_response(response_dict, 200)
+
+# Update the title and description of a task. The User
+# identifies the specific task with the id in the URL.
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_one_task(task_id):
+    task = verify_item(Task, task_id)
+
+    request_body = request.get_json()
+    verify_task_inputs(request_body)
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
```

Comment:
I'd recommend wrapping these lines in a try-except block so that you can return an error message if the user is missing some of the required data